### ISSUE TITLE
cd_history: bad array subscript

### ISSUE THREAD

**Issue by adoyle-h** [spencertipping/cd#16]
It shows many `cd_history: bad array subscript`. I tracked the code and found this.

https://github.com/spencertipping/cd/blob/master/cd-history#L39
https://github.com/spencertipping/cd/blob/master/cd#L71

`"${cd_history[-1]}"` causes the error. I think it is a bug. Could you fix it?

-----

My cd configuration:

```sh
CD_EXTENSIONS=(traverse history)
```

My bash version: GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin15)
And current options for bash:

```
set -o
allexport      	off
braceexpand    	on
emacs          	on
errexit        	off
errtrace       	off
functrace      	on
hashall        	on
histexpand     	off
history        	on
ignoreeof      	on
interactive-comments	on
keyword        	off
monitor        	on
noclobber      	off
noexec         	off
noglob         	off
nolog          	off
notify         	off
nounset        	off
onecmd         	off
physical       	off
pipefail       	on
posix          	off
privileged     	off
verbose        	off
vi             	off
xtrace         	off
```

```
shopt
cdable_vars    	off
cdspell        	off
checkhash      	off
checkwinsize   	on
cmdhist        	on
compat31       	off
dotglob        	off
execfail       	off
expand_aliases 	on
extdebug       	on
extglob        	on
extquote       	on
failglob       	off
force_fignore  	on
gnu_errfmt     	off
histappend     	on
histreedit     	on
histverify     	off
hostcomplete   	on
huponexit      	off
interactive_comments	on
lithist        	off
login_shell    	on
mailwarn       	off
no_empty_cmd_completion	off
nocaseglob     	off
nocasematch    	off
nullglob       	off
progcomp       	on
promptvars     	on
restricted_shell	off
shift_verbose  	off
sourcepath     	on
xpg_echo       	off
```

**Comment by spencertipping** [comment]
Thanks for the bug report! Just pushed a possible fix, though this isn't something I can reproduce easily since bash 3.X is much older than the versions that ship with Linux.

Let me know if the latest works; otherwise I can try to do some digging and see if I can get bash 3.X installed in a docker image.

**Comment by spencertipping** [comment]
...just verified the fix on CentOS 5, bash 3.2 -- optimistically closing the issue. Thanks again for filing this!
